app/Http/Controllers/Auth/LoginController.php

```diff
@@ -22,89 +20,64 @@ class LoginController extends BaseController
 {
     use AuthenticatesUsers;
 
-    /**
-     * Where to redirect users after login.
-     *
-     * @var string
-     */
-    protected $redirectTo = '/mship/manage/dashboard';
+    protected $provider;
 
     /**
      * Create a new controller instance.
      *
      * @return void
      */
-    public function __construct()
-    {
-        $this->middleware('guest')->except('showLoginForm', 'loginSecondary', 'logout');
-    }
-
-    public function getLogin()
+    public function __construct(VATSIMOAuthProvider $provider)
     {
-        if (Auth::guard('vatsim-sso')->check() && ! Auth::check()) {
-            return $this->attemptSecondaryAuth();
-        } else {
-            return redirect()->route('dashboard');
-        }
+        $this->provider = $provider;
     }
 
-    /**
-     * Handle a login request to the application.
-     *
-     * @param  \Illuminate\Http\Request $request
-     * @return \Illuminate\Http\RedirectResponse|\Illuminate\Http\Response
-     */
-    public function loginMain(Request $request)
+    public function login(Request $request)
     {
+        if (! $request->has('code') || ! $request->has('state')) {
+            $authorizationUrl = $this->provider->getAuthorizationUrl();
+            $request->session()->put('vatsimauthstate', $this->provider->getState());
 
-        // user has not been authenticated with VATSIM SSO
-        if (! Auth::guard('vatsim-sso')->check()) {
-            return $this->attemptVatsimAuth();
+            return redirect()->away($authorizationUrl);
         }
 
-        if (! Auth::check()) {
-            $this->attemptSecondaryAuth();
+        if ($request->input('state') !== session()->pull('vatsimauthstate')) {
+            return redirect()->route('dashboard')->withError('Something went wrong, please try again.');
```

Comment:
The user won't have access to dashboard here (I think) - so this will just redirect somewhere else? For efficiency, we should redirect direct to a page they have access to

Reply:
Dashboard is available without auth, it is this page

<img width="1016" alt="image" src="https://user-images.githubusercontent.com/7726792/98736408-ab418400-239c-11eb-8896-5070ccb98fd1.png">